app/controllers/api/storage.php

```diff
@@ -21,8 +22,106 @@
+    ->param('write', null, new ArrayList(new Text(64)), 'An array of strings with write permissions. By default only the current user is granted with write permissions. [learn more about permissions](/docs/permissions) and get a full list of available permissions.', true)
+    ->inject('response')
+    ->inject('dbForInternal')
+    ->inject('user')
+    ->inject('audits')
+    ->action(function ($name, $maximumFileSize, $allowedFileExtensions, $enabled, $adapter, $encryption, $antiVirus, $read, $write, $response, $dbForInternal, $user, $audits) {
+        /** @var Utopia\Swoole\Request $request */
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Utopia\Database\Database $dbForInternal */
+        /** @var Appwrite\Database\Document $user */
+        /** @var Appwrite\Event\Event $audits */
+
+        try {
+            $data = [
+                '$collection' => 'buckets',
+                'dateCreated' => \time(),
+                'dateUpdated' => \time(),
+                'name' => $name,
+                'maximumFileSize' => $maximumFileSize,
+                'allowedFileExtensions' => $allowedFileExtensions,
+                'enabled' => $enabled,
+                'adapter' => $adapter,
+                'encryption' => $encryption,
+                'antiVirus' => $antiVirus,
+            ];
+
+            $data['$read'] = (is_null($read) && !$user->isEmpty()) ? ['user:' . $user->getId()] : $read ?? []; //  By default set read permissions for user
+            $data['$write'] = (is_null($write) && !$user->isEmpty()) ? ['user:' . $user->getId()] : $write ?? []; //  By default set write permissions for user
+            $data = $dbForInternal->createDocument('buckets', new Document($data));
+        } catch (AuthorizationException $exception) {
+            throw new Exception('Unauthorized permissions', 401);
+        } catch (StructureException $exception) {
+            throw new Exception('Bad structure. ' . $exception->getMessage(), 400);
+        } catch (\Exception$exception) {
+            throw new Exception('Failed saving document to DB', 500);
+        }
+
+        $audits
+            ->setParam('event', 'database.collections.create')
+            ->setParam('resource', 'database/collection/' . $data->getId())
+            ->setParam('data', $data->getArrayCopy())
+        ;
+
+        $response->setStatusCode(Response::STATUS_CODE_CREATED);
+        $response->dynamic2($data, Response::MODEL_BUCKET);
+    });
+
+App::get('/v1/storage/buckets')
+    ->desc('List buckets')
+    ->groups(['api', 'storage'])
+    ->label('scope', 'buckets.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_SESSION, APP_AUTH_TYPE_KEY, APP_AUTH_TYPE_JWT])
+    ->label('sdk.namespace', 'storage')
+    ->label('sdk.method', 'listBuckets')
+    ->label('sdk.description', '/docs/references/storage/list-buckets.md')
```

Comment:
We also need to create this description file, which should follow the same line of other similar routes.

Reply:
done